### PR TITLE
Move over more methods from KeyTable to Table

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/api2/table.py [hunk 1]
```diff
@@ -1244,3 +1277,420 @@ def format_type(typ):
...
+        --------
+
+        Test whether `C1` is equal to 5 any row in any row of the table:
+
+        >>> if table1.exists(table1.C1 == 5):
+        ...     print("At least one row has C1 equal 5.")
+
+        :param str expr: Boolean expression.
+
+        :rtype: bool
```
Comment (patrick-schultz) [inline python/hail/api2/table.py]: This should be
```
Parameters
----------
...
Returns
-------
...
```
no?

Comment (tpoterba) [reply python/hail/api2/table.py]: ah, yep. I copied this and didn't edit the docs.

File: python/hail/api2/table.py [hunk 2]
```diff
@@ -1244,3 +1277,420 @@ def format_type(typ):
...
+    @typecheck_method(right=table_type,
+                      how=enumeration('inner', 'outer', 'left', 'right'))
+    def join(self, right, how='inner'):
+        """Join two tables together.
+
+        Examples
+        --------
+        Join `table1` to `table2` to produce `table_joined`:
+
+        >>> table_result = table1.key_by('ID').join(table2.key_by('ID'))
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/api2/table.py]: good catch.

**Comment by tpoterba** [comment]
Depends on bug fix in #2620

**Pull request merged**